pkg/controller/jobs/rayjob/rayjob_webhook.go

```diff
@@ -142,11 +154,6 @@ func (w *RayJobWebhook) validateCreate(ctx context.Context, job *rayv1.RayJob) (
 		clusterSpec := spec.RayClusterSpec
 		clusterSpecPath := specPath.Child("rayClusterSpec")
 
-		// Should not use auto scaler. Once the resources are reserved by queue the cluster should do its best to use them.
-		if ptr.Deref(clusterSpec.EnableInTreeAutoscaling, false) {
```

Comment:
I think we should still make the validation, and allow for `EnableInTreeAutoscaling` only if ElasticJobsViaWorkloadSlices are enabled.

So I would say:

`if ptr.Deref(clusterSpec.EnableInTreeAutoscaling, false) && jobframework.WorkloadSliceEnabled(j) {`

Reply:
Updated